File: conversions/csharp_unity/Narrative.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Narrative : BaseElement
{
    // Nature
    public string History;
    public string Consequences;
    public int? StartDate;
    public int? EndDate;
    // Involves
    public List<string> EventsIds; = new List<string>();
    public List<string> CharactersIds; = new List<string>();
    public List<string> ObjectsIds; = new List<string>();
    public List<string> LocationsIds; = new List<string>();
    public List<string> SpeciesIds; = new List<string>();
    public List<string> CreaturesIds; = new List<string>();
    public List<string> InstitutionsIds; = new List<string>();
    public List<string> TraitsIds; = new List<string>();
    public List<string> CollectivesIds; = new List<string>();
    public List<string> TerritoriesIds; = new List<string>();
    public List<string> AbilitiesIds; = new List<string>();
    public List<string> PhenomenaIds; = new List<string>();
    public List<string> LanguagesIds; = new List<string>();
    public List<string> FamiliesIds; = new List<string>();
    public List<string> RelationsIds; = new List<string>();
    public List<string> TitlesIds; = new List<string>();
    public List<string> ConstructsIds; = new List<string>();
}

File: conversions/typescript/Narrative.ts
import { BaseElement } from './base_elements';

export interface Narrative extends BaseElement {
  // Nature
  history?: string | null;
  consequences?: string | null;
  startDate?: number | null;
  endDate?: number | null;
  // Involves
  eventsIds?: string[] | null;
  charactersIds?: string[] | null;
  objectsIds?: string[] | null;
  locationsIds?: string[] | null;
  speciesIds?: string[] | null;
  creaturesIds?: string[] | null;
  institutionsIds?: string[] | null;
  traitsIds?: string[] | null;
  collectivesIds?: string[] | null;
  territoriesIds?: string[] | null;
  abilitiesIds?: string[] | null;
  phenomenaIds?: string[] | null;
  languagesIds?: string[] | null;
  familiesIds?: string[] | null;
  relationsIds?: string[] | null;
  titlesIds?: string[] | null;
  constructsIds?: string[] | null;
}
